aiida_common_workflows/workflows/relax/generator.py

```diff
@@ -68,28 +85,49 @@ def get_builder(
         threshold_forces=None,
         threshold_stress=None,
         previous_workchain=None,
+        is_insulator=False,
+        spin=SpinType.NONE,
+        initial_magnetization='auto',
         **kwargs
     ):
         """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
 
-        :param structure: the structure to be relaxed
-        :param calc_engines: ...
-        :param protocol: the protocol to use when determining the workchain inputs
-        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`
+        :param structure: the structure to be relaxed.
+        :param calc_engines: a dictionary containing the computational resources for the relaxation.
+        :param protocol: the protocol to use when determining the workchain inputs.
+        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`.
         :param threshold_forces: target threshold for the forces in eV/Å.
         :param threshold_stress: target threshold for the stress in eV/Å^3.
         :param previous_workchain: a <Code>RelaxWorkChain node.
+        :param is_insulator: a bool to activate insulator options (by default metal calcs are assumed.
+        :param spin: the spin state of the calculation, instance of `RelaxType`.
+        :param initial_magnetization: a list with the spin polarization for each site.
         :param kwargs: any inputs that are specific to the plugin.
         :return: a `aiida.engine.processes.ProcessBuilder` instance ready to be submitted.
         """
-        if previous_workchain:
+        if previous_workchain is not None:
             try:
                 prev_wc_class = previous_workchain.process_class
                 if not prev_wc_class == self.process_class:
                     raise ValueError('The "previous_workchain" must be a node of {}'.format(self.process_class))
             except AttributeError:
                 raise ValueError('The "previous_workchain" must be a node of {}'.format(self.process_class))
 
+        if relaxation_type not in self._relaxation_types:
+            raise ValueError('relaxation type `{}` is not supported'.format(relaxation_type))
+
+        if is_insulator not in [False, True, None]:
+            raise ValueError('The argument `is_insulator` accepts only `False`, `True` or `None`')
```

Comment:
If we accept `None`, what is that supposed to mean? What is the difference with `False`? If we do accept this value, maybe that should be the default. Or maybe we should also switch this flag to an enum like `spin`, something like `electronic_type` with an `ElectronicType` enum with for now the fields `INSULATOR`, `METAL`. This will give us more flexibility in the future to extend it without risking to having to break the interface

Reply:
None signal cases when the implementation can automatically decide whether to run a calculation with metal option or not. I suggested "auto" during last meeting, but in order to not mix string you suggested None. Because not all implementations can automatically decide whether to run a calculation with metal option, the default must be False, not None. I don't see the reason to have an `ElectronicType`, there aren't more options to cover I guess. Do you see some?

Reply:
well there is `INSULATOR`, `METAL` and `NONE`. The `NONE` is a real third option in this context as in this means that the workflow will have to try and determine it automatically which is very different from the other two. I think having it as an explicit value makes it a lot more intuitive and safe that `None`. I may have suggested this, but only as an improvement over using a string. On top of that, could it be that in the future (some) plugins may want or need to specify for example `SEMI_METAL` or `SEMI_CONDUCTOR`? It may be a stretch, but by using a type, we allow for this flexibility in the future and don't have to break the interface, which would cause a lot of problems. Finally, since we already use enums for `RelaxType` and `SpinType`, I think having the same interface for `ElectronicType` makes sense and will give a homogeneous interface

Reply:
Ok with me. Only thing is that this decision is deviating a bit from what we agreed at the last meeting. Shouldn't be a problem, but maybe let's leave few hours to allow others to express their opinion in case. @giovannipizzi @espenfl @sponce24 @chrisjsewell @ConradJohnston @eimrek @pzarabadip @yakutovicha @adegomme @broeder-j

Reply:
OK with me to have multiple values.
In this case, however, rather than `NONE` I would use something that really says what's going to happen, like `AUTOMATIC`? `NONE` is a bit unclear.

Reply:
I agree with @giovannipizzi I also prefer 'AUTOMATIC' than 'NONE'

Reply:
fine for me to make it `AUTOMATIC`.

Reply:
Thanks for the feedback. I implemented the enum ElectronicType and the logic related to it